Don't create new numpy arrays when inserting in the DataTree

The function `_chexpand()` creates new numpy objects. This is a problem when I want to insert `sparse.COO` objects into the tree, as it automatically expands to a full dense numpy array and therefore I get an error, because `sparse` is blocking that by default. Also, I wouldn't want it to happen.

```py
    def __array__(self, *args, **kwargs):
        from ._settings import AUTO_DENSIFY
    
        if not AUTO_DENSIFY:
>           raise RuntimeError(
                "Cannot convert a sparse array to dense automatically. To manually densify, use the todense method."
            )
E           RuntimeError: Cannot convert a sparse array to dense automatically. To manually densify, use the todense method.
```

It's this piece of code:

https://github.com/modflowpy/xattree/blob/cab7d2d99ed5aac4dbb774a2483ef77fa2fe59cd/xattree/__init__.py#L314-L320

--- a/xattree/__init__.py
+++ b/xattree/__init__.py
@@ -318,13 +318,13 @@ def _chexpand(value: ArrayLike, shape: tuple[int]) -> NDArray:
     """
 
     try:
-        shp = value.shape
+        shp = value.shape # type: ignore
     except AttributeError:
         return np.full(shape, value)
     except Exception:
-        raise ValueError(f"Unsupported array item: {value}")
+        raise ValueError(f"Unsupported array item type : {type(value)}")
     if shp == ():
-        return np.full(shape, value.item())
+        return np.full(shape, value.item()) # type: ignore
     if shp != shape:
         raise ValueError(f"Shape mismatch, got {shp}, expected {shape}")
     # any way to avoid the cast?

--- a/xattree/__init__.py
+++ b/xattree/__init__.py
@@ -316,18 +316,18 @@ def _chexpand(value: ArrayLike, shape: tuple[int]) -> NDArray:
     Check an array-like value's shape. If it's a scalar, expand it
     to the requested shape. If an array, make sure it's that shape.
     """
+
     try:
-        shp = value.shape  # type: ignore
-    except:  # noqa
-        try:
-            value = np.asanyarray(value)
-            shp = value.shape
-        except:  # noqa
-            return np.full(shape, value)
-    if shp == ():
+        shp = value.shape
+    except AttributeError:
         return np.full(shape, value)
+    except Exception:
+        raise ValueError(f"Unsupported array item: {value}")
+    if shp == ():
+        return np.full(shape, value.item())
     if shp != shape:
         raise ValueError(f"Shape mismatch, got {shp}, expected {shape}")
+    # any way to avoid the cast?
     return cast(NDArray, value)
 
 

--- a/xattree/__init__.py
+++ b/xattree/__init__.py
@@ -318,13 +318,13 @@ def _chexpand(value: ArrayLike, shape: tuple[int]) -> NDArray:
     """
 
     try:
-        shp = value.shape # type: ignore
+        shp = value.shape  # type: ignore
     except AttributeError:
         return np.full(shape, value)
     except Exception:
         raise ValueError(f"Unsupported array item type : {type(value)}")
     if shp == ():
-        return np.full(shape, value.item()) # type: ignore
+        return np.full(shape, value.item())  # type: ignore
     if shp != shape:
         raise ValueError(f"Shape mismatch, got {shp}, expected {shape}")
     # any way to avoid the cast?

--- a/xattree/__init__.py
+++ b/xattree/__init__.py
@@ -311,13 +311,24 @@ _XTRA_GETTERS = {
 }
 
 
-def _chexpand(value: ArrayLike, shape: tuple[int]) -> Optional[NDArray]:
-    value = np.array(value)
-    if value.shape == ():
-        return np.full(shape, value.item())
-    if value.shape != shape:
-        raise ValueError(f"Shape mismatch, got {value.shape}, expected {shape}")
-    return value
+def _chexpand(value: ArrayLike, shape: tuple[int]) -> NDArray:
+    """
+    Check an array-like value's shape. If it's a scalar, expand it
+    to the requested shape. If an array, make sure it's that shape.
+    """
+    try:
+        shp = value.shape  # type: ignore
+    except:  # noqa
+        try:
+            value = np.asanyarray(value)
+            shp = value.shape
+        except:  # noqa
+            return np.full(shape, value)
+    if shp == ():
+        return np.full(shape, value)
+    if shp != shape:
+        raise ValueError(f"Shape mismatch, got {shp}, expected {shape}")
+    return cast(NDArray, value)
 
 
 @define

